app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt

```diff
@@ -370,6 +382,18 @@ class TabsTrayFragment : AppCompatDialogFragment() {
         }
     }
 
+    @VisibleForTesting
+    internal fun showCancelledDownloadWarning(arg: CancelDownloadFragmentArguments) {
```

Comment:
After looking at this some more, it's kinda weird to wrap all the arguments into a data class and then unbundle it here. I would consider just dropping `CancelDownloadFragmentArguments`. I assume you were maybe doing this because callback functions that are being passed from here into the controller.